solidity/test/GovernanceTest.js

```diff
@@ -180,55 +185,139 @@ describe("TBTCSystem governance", async function() {
+      await tbtcSystem.beginKeepVendorSingleShotUpdate(
         "0x0000000000000000000000000000000000000001",
-        "0x0000000000000000000000000000000000000002",
+        newKeepVendor.address,
       )
 
-      const finalizationTime = await tbtcSystem.getRemainingKeepFactorySingleShotUpdateTime()
+      const finalizationTime = await tbtcSystem.getRemainingKeepVendorSingleShotUpdateTime()
       await increaseTime(finalizationTime.toNumber() + 1) // 10 days
-      await tbtcSystem.finalizeKeepFactorySingleShotUpdate()
+      await tbtcSystem.finalizeKeepVendorSingleShotUpdate()
 
       await expectRevert(
-        tbtcSystem.finalizeKeepFactorySingleShotUpdate(),
+        tbtcSystem.finalizeKeepVendorSingleShotUpdate(),
         "Change not initiated",
       )
     })
   })
 
+  describe("when trying to lock Keep factories version updates", async () => {
+    beforeEach(async () => {
+      await createSnapshot()
+    })
+
+    afterEach(async () => {
+      await restoreSnapshot()
+    })
+
+    it("call completes and emits an event", async () => {
+      const ethStakedFactory = newKeepFactory
+      const ethStakedVendor = newKeepVendor
+
+      // Set ETH-staked vendor and Keep factory selector
+      await tbtcSystem.beginKeepVendorSingleShotUpdate(
+        ecdsaKeepFactory.address,
+        ethStakedVendor.address,
+      )
+      const finalizationTime = await tbtcSystem.getRemainingKeepVendorSingleShotUpdateTime()
+      await increaseTime(finalizationTime.addn(1)) // 10 days
+      await tbtcSystem.finalizeKeepVendorSingleShotUpdate()
+
+      const receipt = await tbtcSystem.lockKeepFactoriesVersionsUpdates(
+        ecdsaKeepFactory.address,
+        ethStakedFactory.address,
+      )
+
+      expectEvent(receipt, "KeepFactoriesVersionsLocked", {
+        _keepStakeFactory: ecdsaKeepFactory.address,
+        _fullyBackedFactory: ethStakedFactory.address,
+      })
+    })
+
+    it("reverts when lockKeepFactoriesVersionsUpdates is called twice", async () => {
+      const ethStakedFactory = newKeepFactory
+      const ethStakedVendor = newKeepVendor
+
+      // Set ETH-staked vendor and Keep factory selector
+      await tbtcSystem.beginKeepVendorSingleShotUpdate(
+        ecdsaKeepFactory.address,
+        ethStakedVendor.address,
+      )
+      const finalizationTime = await tbtcSystem.getRemainingKeepVendorSingleShotUpdateTime()
+      await increaseTime(finalizationTime.addn(1)) // 10 days
+      await tbtcSystem.finalizeKeepVendorSingleShotUpdate()
+
+      await tbtcSystem.lockKeepFactoriesVersionsUpdates(
+        ecdsaKeepFactory.address,
+        ethStakedFactory.address,
+      )
+
+      await expectRevert(
+        tbtcSystem.lockKeepFactoriesVersionsUpdates(
+          ecdsaKeepFactory.address,
+          ethStakedFactory.address,
+        ),
+        "Already locked",
+      )
+    })
+
+    it("reverts when lockKeepFactoriesVersionsUpdates is by non-owner", async () => {
```

Comment:
Maybe just `reverts when not performed by owner`?

Reply:
Wanted to be consistent with other names in this file. I'll update it.

Reply:
d9ac37d0d67048873e2cd54f7b6227b309178069